index.ts

```diff
@@ -199,13 +199,6 @@ export default class Analytics {
     message: any,
     callback: AnalyticsMessageCallback = () => {}
   ): void {
-    if (this.queue.length >= this.maxInternalQueueSize) {
```

Comment:
this should never happen. queue length is measured by the messages in the queue, queueSize is measured by the byte size of the combined message totals. I can't imagine a scenario where the queue has >= 32k messages

Reply:
that's a really good point. after checking the respective libraries:

- rudderstack is setting caps based on the number of messages in the queue 
- segment is setting caps based on size of all messages in the queue combined

To me, it seems we want to both honor the 4 MiB server limit and provide some hard cap at which we stop caching events.

IMO, honoring the 4 MiB server limit would be validating that the size of the upcoming flush is less than 4mb, and if it isn't, flushing fewer events total. Ensuring that we stop caching events after a certain point could be done either based on total queue size or just events in the queue. I'm more partial to using the event totals, it's simpler and I don't think we'd be introducing a point of failure even though the measurement is less precise.

Reply:
Another thing we could do is count the bytes only when creating the batch to send to the server.

Reply:
i like it